File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int cnt = Integer.parseInt(br.readLine());
		StringBuilder sb = new StringBuilder();
		Stack<Integer> stack = new Stack<>();
		int result = 0;
		
		for (int i = 0; i < cnt; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String s = st.nextToken();
            switch (s) {
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    stack.add(num);
                    result = num;
                    break;
                case "pop":
                    if (stack.isEmpty()) {
                        sb.append("-1\n");
                    } else {
                        sb.append(stack.pop() + "\n");
                    }
                    break;
                case "size":
                    sb.append(stack.size() + "\n");
                    break;
                case "empty":
                    if (stack.isEmpty()) sb.append("1\n");
                    else sb.append("0\n");
                    break;
                case "top":
                    if (stack.isEmpty()) sb.append("-1\n");
                    else sb.append(stack.peek() + "\n");
                    break;
            }
        }
        System.out.println(sb);

	}

}

File: 백준/Silver/10828. 스택/스택.kt
import java.util.Stack
import java.util.StringTokenizer

fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt() // 명령의 수
    val ans = StringBuilder()
    val stack = Stack<Int>()

    for (i in 1..n) {
        val line = StringTokenizer(readLine())
        val order = line.nextToken() // 명령

        // push X: 정수 X를 스택에 넣는 연산
        if (order == "push") {
            stack.push(line.nextToken().toInt())
        }
        else {
            val result = when (order) {
                "pop" -> if (stack.isEmpty()) -1 else stack.pop()
                "size" -> stack.size
                "empty" -> if (stack.isEmpty()) 1 else 0
                else -> if (stack.isEmpty()) -1 else stack.peek()
            }

            ans.append(result).append("\n")
        }

    }

    println(ans)
}